### PR TITLE
EVAKA-4190 Unify message access control

### PR DESCRIPTION
**Pull request by akheron**
#### Summary

Reduce copy-paste when checking for message account access.

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageController.kt [hunk 1]
```diff
@@ -185,9 +171,8 @@ class MessageController(
         @RequestBody body: ReplyToMessageBody,
     ) {
         Audit.MessagingReplyToMessageWrite.log(targetId = messageId)
-        authorizeAllowedMessagingRoles(user)
-        val account = db.read { it.getMessageAccountsForEmployee(user) }.find { it.id == accountId }
-            ?: throw Forbidden("Message account not found for user")
+        requireMessageAccountAccess(db, user, accountId)
+        val account = requireMessageAccountAccess(db, user, accountId)
```
Comment (kimifoud) [inline service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageController.kt:175]: whoops
```suggestion
        val account = requireMessageAccountAccess(db, user, accountId)
```

**Review by kimifoud — CHANGES_REQUESTED** [review]

File: service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageController.kt (hunk 1 above)
Comment (akheron) [reply service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageController.kt]: 🤦

Comment (akheron) [reply service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageController.kt]: Fixed

**Review by kimifoud — APPROVED** [review]
👍

**Pull request merged**